atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/SweepStrategyManagers.java

```diff
@@ -33,13 +35,15 @@ private SweepStrategyManagers() {
     }
 
     public static SweepStrategyManager createDefault(KeyValueService kvs) {
-        RecomputingSupplier<Map<TableReference, SweepStrategy>> supplier =
-                RecomputingSupplier.create(() -> getSweepStrategies(kvs));
+        // On a cache miss, load metadata only for the relevant table. Helpful when many dynamic tables.
+        LoadingCache<TableReference, SweepStrategy> cache = Caffeine.newBuilder()
+                .build(tableRef -> getSweepStrategy(kvs.getMetadataForTable(tableRef)));
         return tableRef -> {
-            SweepStrategy strategy = supplier.get().get(tableRef);
-            if (strategy == null) {
-                strategy = supplier.recompute().get(tableRef);
+            // Add all existing tables the first time this is called, to optimize for cases when mostly non-dynamic tables.
+            if (cache.estimatedSize() == 0) {
+                cache.putAll(getSweepStrategies(kvs));
```

Comment:
in this initial case doesn't this degrade to the version on the left on initial load? or is that an acceptable compromise?

Reply:
If you are asking if it's okay to load all tables a single time: Even before https://github.com/palantir/atlasdb/pull/4542, we loaded the sweep strategy for all tables initially, and hadn't heard complaints around perf. So my _guess_ is this is just a problem when dynamic tables get created often, and thus now we limit loading all tables to only happen once.

If your concern is instead about how we no longer clear deleted table entries from the cache after seeing a new table entry: I hope that's fine.

Reply:
There's still a potential problem here where if loading all the sweep strategies takes a long time, you could have many threads all hit this case (as the cache size would still be 0 while the first thread is still loading).

Reply:
hmm yeah, might we want to have a `Suppliers.memoize` on line 42?

Reply:
If memoizing, we'd want that to happen outside of the lambda, right?

Reply:
oop I can't read lol, but yes

Reply:
Pushed https://github.com/palantir/atlasdb/pull/4625